temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowThreadContext.java

```diff
@@ -318,7 +312,11 @@ public boolean isDestroyRequested() {
     }
   }
 
-  public void destroy() {
+  /**
+   * Non-blocking call, never throws.<br>
+   * There is no guarantee that the thread is destroyed at the end of this call.
+   */
+  void destroy() {
```

Comment:
This should maybe be called tryDestroy or something since it's not guaranteed to actually destroy

Reply:
It does guarantee, it just doesn't block on it. I will rename to `initiateDestroy`.